pkg/controllers/provisioning/scheduling/topologygroup.go

```diff
@@ -210,6 +210,19 @@ func (t *TopologyGroup) nextDomainTopologySpread(pod *corev1.Pod, podDomains, no
 	minDomain := ""
 	minCount := int32(math.MaxInt32)
 
+	// We special-case kubernetes.io/hostname primarily for new NodeClaims since their domain won't be registered until we Add() them
+	if t.Key == corev1.LabelHostname && len(nodeDomains.Values()) == 1 {
+		hostName := nodeDomains.Values()[0]
+		count := t.domains[hostName] // t.domains[hostName] produces a 0 value for new NodeClaims
+		if selfSelecting {
+			count++
+		}
+		if count <= t.maxSkew {
+			return scheduling.NewRequirement(t.Key, corev1.NodeSelectorOpIn, hostName)
+		}
```

Comment:
I think I get this now, but let's double check my understanding. This works because we assume we can always create additional hosts, so the maximum number of pods we should attempt to schedule to a node is the max skew since we can treat the global minimum as zero. If I've got that right I think it's worth a comment since it's not how Kubernetes reasons about maxSkew in general, it's just how Karpenter reasons about it since we can always create an additional domain.

Reply:
Yeah, you've got that right -- I can add a comment -- good call